file_store/src/file_info_poller.rs

```diff
@@ -408,23 +467,208 @@ impl FileInfoPollerState for sqlx::Pool<sqlx::Postgres> {
             .map_err(Error::from)
     }
 
-    async fn clean(&self, process_name: &str, file_type: &str) -> Result {
-        sqlx::query(
+    async fn clean(
+        &self,
+        process_name: &str,
+        file_type: &str,
+        offset: DateTime<Utc>,
+    ) -> Result<u64> {
+        let t100_timestamp: Option<DateTime<Utc>> = sqlx::query_scalar(
+            r#"
+                SELECT file_timestamp
+                FROM files_processed
+                WHERE process_name = $1
+                    AND file_type = $2
+                ORDER BY file_timestamp DESC
+                LIMIT 1 OFFSET 100;
```

Comment:
Do we actually care about the 100 limit now, is finding the above timestamp of any value ?  Why not eliminate that and purge anything outside of the offset window?

Reply:
Depending on the how often a given service processes files, keeping around some amount of history past the offset window allows the offset window to be increased while potentially forgetting to provide a new `StartAfter` lookback value. 

I don't think it's a great foot gun shield. But it is an added benefit along with more history during debugging.